pkg/storage/stores/indexshipper/compactor/deletion/request_handler.go

```diff
@@ -81,6 +90,63 @@ func (dm *DeleteRequestHandler) AddDeleteRequestHandler(w http.ResponseWriter, r
 	w.WriteHeader(http.StatusNoContent)
 }
 
+func shardDeleteRequestsByQueryRange(startTime, endTime model.Time, query, userID string, queryRange time.Duration) []DeleteRequest {
+	var deleteRequests []DeleteRequest
```

Comment:
nit: small optimization
```suggestion
	deleteRequests := make([]DeleteRequest, 0, (endTime.Sub(startTime))/queryRange))
```